Mathlib/Algebra/Quotient.lean

```diff
@@ -0,0 +1,61 @@
+/-
+Copyright (c) 2021 Anne Baanen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Anne Baanen
+-/
+import Mathlib.Mathport.Rename
+/-!
+# Algebraic quotients
+
+This file defines notation for algebraic quotients, e.g. quotient groups `G ⧸ H`,
+quotient modules `M ⧸ N` and ideal quotients `R ⧸ I`.
+
+The actual quotient structures are defined in the following files:
+ * quotient group: `Mathlib/GroupTheory/QuotientGroup.lean`
+ * quotient module: `Mathlib/LinearAlgebra/Quotient.lean`
+ * quotient ring: `Mathlib/RingTheory/Ideal/Quotient.lean`
+
+## Notations
+
+The following notation is introduced:
+
+* `G ⧸ H` stands for the quotient of the type `G` by some term `H`
+  (for example, `H` can be a normal subgroup of `G`).
+  To implement this notation for other quotients, you should provide a `has_quotient` instance.
+  Note that since `G` can usually be inferred from `H`, `_ ⧸ H` can also be used,
+  but this is less readable.
+
+## Tags
+
+quotient, group quotient, quotient group, module quotient, quotient module, ring quotient,
+ideal quotient, quotient ring
+-/
+
+
+universe u v
+
+/-- `has_quotient A B` is a notation typeclass that allows us to write `A ⧸ b` for `b : B`.
```

Comment:
```suggestion
/-- `HasQuotient A B` is a notation typeclass that allows us to write `A ⧸ b` for `b : B`.
```

Reply:
I think the class itself should be called `Quotient` according to the naming convention?

Reply:
It would, but that name is already taken. There are a few notation typeclasses in core which also have this issue: `AndOp` and `OrOp` are the names of the notation typeclasses for bitwise and / or, which have to avoid clashing with the propositions `And` and `Or`. So `QuotientOp` is another option. But `HasQuotient` seems like the least disruptive option.